Contents/WeaponModification/WeaponModificationUI.cs

```diff
@@ -348,6 +296,94 @@ public ImprovisedUITextBox(string text, float textScale = 1f, bool large = false
+				return;
+			Player player = Main.player[whoAmI];
+			if (player.TryGetModPlayer(out WeaponModificationPlayer modplayer)) {
+				this.player = player;
+				Vector2 originDefault = new Vector2(26, 26);
+				int maxLengthX = 550;
+				WeaponModificationWeaponUISlot wpUI = new WeaponModificationWeaponUISlot(TextureAssets.InventoryBack2, player);
+				wpUI.UISetPosition(player.Center + new Vector2(100, 40), originDefault);
+				Append(wpUI);
+				for (int i = 0; i < modplayer.WeaponModification_inventory.Length; i++) {
+					Vector2 offset = new Vector2(MathHelper.Lerp(-maxLengthX, maxLengthX, i / (modplayer.WeaponModification_inventory.Length - 1f)), 100);
+					Vector2 position = player.Center + offset;
+					if (i >= (modplayer.WeaponModification_inventory.Length - 1) * .5f) {
+						position -= new Vector2(maxLengthX, -55);
+					}
+					WeaponModificationUIslot inventory = new WeaponModificationUIslot(TextureAssets.InventoryBack, player);
+					inventory.WhoAmI = i;
+					inventory.ModificationType = modplayer.WeaponModification_inventory[i];
+					inventory.UISetWidthHeight(52, 52);
+					inventory.UISetPosition(position, originDefault);
+					Append(inventory);
+				}
+			}
+			DeletionSlot delslot = new DeletionSlot(TextureAssets.InventoryBack11, player);
+			delslot.UISetWidthHeight(52, 52);
+			delslot.UISetPosition(player.Center + new Vector2(-580, 155), new Vector2(26, 26));
+			Append(delslot);
+		}
+		public override void OnDeactivate() {
+			WeaponModificationSystem.SelectedInventorySlot = -1;
+			WeaponModificationSystem.SelectedModifySlot = -1;
+			if (player == null)
+				return;
+			if (player.TryGetModPlayer(out WeaponModificationPlayer modplayer)) {
+				foreach (var element in Elements) {
+					if (element is WeaponModificationUIslot wmUISlot && wmUISlot != null) {
+						if (wmUISlot.WeaponModificationType != null) {
+							if (wmUISlot.item == null) {
+								modplayer.WeaponModification_inventory[wmUISlot.WhoAmI] = (int)wmUISlot.WeaponModificationType;
+							}
+						}
+					}
+				}
+			}
+		}
+		public override void LeftMouseDown(UIMouseEvent evt) {
+			base.LeftMouseDown(evt);
+		}
+	}
+	public class DeletionSlot : UIImage {
+		Player player;
+		Texture2D texture;
+		public DeletionSlot(Asset<Texture2D> texture, Player player) : base(texture) {
+			this.player = player;
+			this.texture = texture.Value;
+		}
+		public override void LeftClick(UIMouseEvent evt) {
+			if (WeaponModificationSystem.SelectedInventorySlot != -1) {
+				player.GetModPlayer<WeaponModificationPlayer>().WeaponModification_inventory[WeaponModificationSystem.SelectedInventorySlot] = -1;
+				WeaponModificationSystem.SelectedInventorySlot = -1;
+			}
+			if (WeaponModificationSystem.SelectedModifySlot != -1) {
+				WeaponModificationUIslot UIslot = (WeaponModificationUIslot)Parent.Children.Where(e => e is WeaponModificationUIslot { item: not null } slot && slot.WhoAmI == WeaponModificationSystem.SelectedModifySlot).FirstOrDefault();
```

Comment:
For the sake of readability you should split this long line into multiple lines. For example:
```
var UIslot = Parent.Children
  .OfType<WeaponModificationUIslot>()
  .Where(...)
  .FirstOrDefault();
```